### ISSUE TITLE
dart2js compiler crashed, dart: 2.0.0-dev.39.0

### ISSUE THREAD

**Issue by rwrozelle** [dart-lang/sdk#32623]
Lubuntu 17.10
4.13.0-37-generic #42-Ubuntu SMP Wed Mar 7 14:13:23 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

$ pub run build_runner build --config release --output build
[INFO] Generating build script completed, took 858ms
[WARNING] Throwing away cached asset graph because the build phases have changed. This most commonly would happen as a result of adding a new dependency or updating your dependencies.
[INFO] Cleaning up outputs from previous builds. completed, took 162ms
[INFO] Reading cached asset graph completed, took 2.6s
[INFO] Building new asset graph completed, took 4.3s
[INFO] Checking for unexpected pre-existing outputs. completed, took 4ms
[INFO] build_web_compilers|entrypoint on web/main.dart:Running dart2js with --dump-info --fast-startup --minify --no-frequency-based-minification --no-source-maps --show-package-warnings --trust-primitives --trust-type-annotations --packages=.package-71073a2061fb92cd26197a9bcf318793 -oweb/main.dart.js web/main.dart
[SEVERE] build_web_compilers|entrypoint on web/main.dart:
Dart2Js finished with:

/b/build/slave/dart-sdk-linux-dev/build/sdk/sdk/lib/_internal/js_runtime/lib/isolate_helper.dart@43652+1:
Internal Error: The compiler crashed when compiling this element.
The compiler is broken.

When compiling the above element, the compiler crashed. It is not
possible to tell if this is caused by a problem in your program or
not. Regardless, the compiler should not crash.

The Dart team would greatly appreciate if you would take a moment to
report this problem at http://dartbug.com/new.

Please include the following information:

the name and version of your operating system,

the Dart SDK build number (2.0.0-dev.39.0), and

the entire message you see here (including the full stack trace
below as well as the source location above).

The compiler crashed: NoSuchMethodError: Class 'JClass' has no instance getter 'superclass'.
Receiver: Instance of 'JClass'
Tried calling: superclass
#0 Object.noSuchMethod (dart:core-patch/dart:core/object_patch.dart:46)
#1 new _ConstructorBodyNamingScope. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/js_backend/minify_namer.dart:342)
#2 _HashVMBase&MapMixin&&_LinkedHashMapMixin.putIfAbsent (dart:collection-patch/dart:collection/compact_hash.dart:277)
#3 new _ConstructorBodyNamingScope (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/js_backend/minify_namer.dart:341)
#4 Namer&_MinifiedFieldNamer&_MinifyConstructorBodyNamer.constructorBodyName (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/js_backend/minify_namer.dart:367)
#5 Namer.instanceMethodName (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/js_backend/namer.dart:790)
#6 SsaCodeGenerator.visitInvokeConstructorBody (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:1771)
#7 HInvokeConstructorBody.accept (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/nodes.dart:1894)
#8 SsaCodeGenerator.visit (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:711)
#9 SsaCodeGenerator.visitStatement (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:723)
#10 SsaCodeGenerator.define (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:692)
#11 SsaCodeGenerator.iterateBasicBlock (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:1373)
#12 SsaCodeGenerator.enterSubGraph (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:1241)
#13 SsaCodeGenerator.visitSubGraph (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:429)
#14 SsaCodeGenerator.visitGraph (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:420)
#15 SsaCodeGeneratorTask.generateMethod. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:118)
#16 CompilerTask.measure (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:63)
#17 SsaCodeGeneratorTask.generateMethod (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:101)
#18 SsaCodeGeneratorTask.generateCode (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/codegen.dart:70)
#19 SsaFunctionCompiler.compile (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/ssa/ssa.dart:47)
#20 JavaScriptBackend.codegen (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/js_backend/backend.dart:848)
#21 KernelCodegenWorkItem.run (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/kernel/kernel_backend_strategy.dart:79)
#22 Compiler.emptyQueue..... (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:784)
#23 CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178)
#24 Compiler.emptyQueue.... (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:784)
#25 CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178)
#26 Compiler.emptyQueue... (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:782)
#27 CompilerDiagnosticReporter.withCurrentElement (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:1156)
#28 Compiler.emptyQueue.. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:780)
#29 EnqueuerStrategy.processWorkItem (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/enqueue.dart:513)
#30 CodegenEnqueuer.forEach (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/js_backend/enqueuer.dart:227)
#31 Compiler.emptyQueue. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:776)
#32 CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178)
#33 Compiler.emptyQueue (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:775)
#34 Compiler.processQueue. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:797)
#35 CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178)
#36 Compiler.processQueue (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:794)
#37 Compiler.compileLoadedLibraries. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:640)
#38 CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178)
#39 Compiler.compileLoadedLibraries (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:534)
#40 Compiler.runInternal (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:488)

#41 Compiler.run.. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:313)
#42 new Future.sync (dart:async/future.dart:222)
#43 Compiler.run. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:313)
#44 CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178)
#45 Compiler.run (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:310)
#46 CompilerImpl.run.. (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/pkg/compiler/lib/src/apiimpl.dart:265)
#47 _RootZone.runUnary (dart:async/zone.dart:1381)
#48 _FutureListener.handleValue (dart:async/future_impl.dart:129)
#49 _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:633)
#50 _Future._propagateToListeners (dart:async/future_impl.dart:662)
#51 _Future._completeWithValue (dart:async/future_impl.dart:477)
#52 _Future._asyncComplete. (dart:async/future_impl.dart:507)
#53 _microtaskLoop (dart:async/schedule_microtask.dart:41)
#54 _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#55 _runPendingImmediateCallback (dart:isolate-patch/dart:isolate/isolate_patch.dart:113)
#56 _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:166)

[INFO] Running build completed, took 2m 25s
[INFO] Caching finalized dependency graph completed, took 862ms
[INFO] Reading manifest at build/.build.manifest completed, took 6ms
[INFO] Deleting previous outputs in build completed, took 357ms
[INFO] Creating merged output dir build completed, took 5.9s
[INFO] Writing asset manifest completed, took 5ms
[INFO] Succeeded after 2m 32s with 2062 outputs

**Comment by Bohne13** [comment]
Use dart: 2.0.0-dev.40.0, could be possible, it's fixed. But no promises!

**Comment by rwrozelle** [comment]
Still broken after update to dart: 2.0.0-dev.40.0.  No luck.

**Comment by sigmundch** [comment]
I landed a fix for this issue last night, but unfortunately it didn't make it on version  2.0.0-dev.40.0, but should make it in version 2.0.0-dev.41.0

The issue comes up only if you are using `--no-frequency-based-minification`, so if you can work for now without that flag, that should hopefully unblock you until the next release (which is expected in a couple days I believe).

I'll mark this as a duplicate of https://github.com/dart-lang/sdk/issues/32600, but let me know if you think there is a different problem at play here.

Cheers!
